docs/onboarding-guide.md

```diff
@@ -0,0 +1,171 @@
+* Add a new launch profile called `AWS.Deploy.CLI`
+* In the *Command line arguments* text box add the following (leave other text boxes blank): 
+
+```
+deploy —project-path .../path/to/aws-dotnet-deploy/testapps/WebAppWithDockerFile
+```
+
+*The above command assumes you have a [default] profile configured in your AWS credentials file. If you want to use a different profile, use the `--profile` switch*
+
+* Select the `AWS.Deploy.CLI` launch profile and run the tool
+* Follow the prompts on the screen and perform a deployment using any compatible recommendation.
+* A successful deployment indicates that your environment is set up correctly and you can proceed to debugging the source code.
+
+## Repository Overview
+
+The .NET deployment tool comprises of various top level components.
+
+
+### [AWS.Deploy.CLI](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI)
+
+This component controls the UI elements for the .NET deployment tool and serves as the first entry point when running the tool. This is the layer that end users directly interact with and it uses the [System.CommandLine](https://docs.microsoft.com/en-us/dotnet/standard/commandline/) package for the CLI UX.
+
+#### Key Components
+
+* [**Program.cs**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Program.cs) - Serves at the entry point to the CLI application.
+* [**CustomServiceCollectionExtension**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Extensions/CustomServiceCollectionExtension.cs) - This serves as the dependency injection container and instantiates all the dependencies required by the deployment tool.
+* [**CommandFactory**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/CommandFactory.cs) - This instantiates and executes the top-level ***dotnet aws***  command and other sub-commands like [*deploy*](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/DeployCommand.cs), [*list-deployments*](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/ListDeploymentsCommand.cs) and [*delete-deployment*](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/DeleteDeploymentCommand.cs)*.* It also holds all the [CLI switches](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/CommandFactory.cs#L40-#L49) that are applicable to the any command. All new commands/switches will be declared and instantiated via this class. 
+
+* **[CommandHandlerInput](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI/Commands/CommandHandlerInput)** - This contains the POCO objects that hold the values for all CLI switches of the associated command. These objects are provided as input while creating the command handlers. See [here](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.CLI/Commands/CommandFactory.cs#L185) for an example.
```

Comment:
We should use links from specific commits. Then people will look at stale versions.

Reply:
I now recognize some of these links are to specific line numbers so you might need the link with commit ids. Please just do this as sparing as possible and if no line number is given then you shouldn't need the commit id.